src/platform/FailSafeContext.cpp

```diff
@@ -76,8 +76,12 @@ void FailSafeContext::FailSafeTimerExpired()
         ChipLogError(DeviceLayer, "Failed to post fail-safe timer expired: %" CHIP_ERROR_FORMAT, status.Format());
     }
 
-    mFailSafeBusy = true;
+    ScheduleFailSafeCleanup();
```

Comment:
Here's the question.  Why can't the PostEvent be inside that function too?

That is, make the function take the fabric index and the two booleans as arguments, call it here with our members, call it in configuration manager with the deserialized state, have the function set mFailSafeArmed/mAddNocCommandHasBeenInvoked/mUpdateNocCommandHasBeenInvoked to false, set mFailSafeBusy to true, post the event, schedule the HandleDisarmFailSafe.

Reply:
Acked